### ISSUE TITLE
websoft9 failed by install srcipt on Alma Linux,Rocky Linux,Oracle Linux,alibabacloudlinux

### ISSUE THREAD

**Issue by qiaofeng1227** [Websoft9/websoft9#555]
```
Package docker-compose-plugin-2.31.0-1.el9.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
[Websoft9] - Docker and Docker Compose installation verified successfully.
install_docker failed with error 0. Exiting.

[Backend] -  Install backend docker services
Set Docker for Websoft9 backend service...
firewalld is not running
install.sh: line 429: cd: /data/websoft9/source/docker: No such file or directory
Failed to change directory.


-------- Systemd --------

[Systemd] -  Install Systemd service
cp: cannot stat '/data/websoft9/source/systemd/script/*': No such file or directory
cp: cannot stat '/data/websoft9/source/systemd/websoft9.service': No such file or directory
Failed to copy Systemd service file.
bash: /data/websoft9/source/install/install_cockpit.sh: No such file or directory
bash: /data/websoft9/source/install/install_plugins.sh: No such file or directory
Restart Docker for Firewalld...
Script execution time: 37 seconds

-- Install success! ------
Access Websoft9 console by: http://Internet IP:9090 and using Linux user for login

```

**Comment by qiuf12** [comment]
```
Ansible自动化安装Appstore...
ansible-playbook -i hosts application.yml -e region=1 -e init=1 -e image=websoft9-basic080645-AlibabaCloudLinux3.2104 -e image_type= -e appname= -e extra_app=desktop -e appnumber=3 -e appinstall=2 -e solution= -e dist=community -e version=basic080645

PLAY [Deploy Application] ***********************************************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************
[WARNING]: Platform linux on host 121.40.237.179 is using the discovered Python interpreter at /usr/bin/python3.6, but future installation of another Python
interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/2.16/reference_appendices/interpreter_discovery.html for more
information.
ok: [121.40.237.179]

TASK [Show install app] *************************************************************************************************************************************
ok: [121.40.237.179] => {
    "msg": "start to install -community-basic080645"
}

TASK [Delete all applications and websoft9 data for install again] ******************************************************************************************
changed: [121.40.237.179]

TASK [Install tools(wget,jq)] *******************************************************************************************************************************
changed: [121.40.237.179]

TASK [Preparing to start installation] **********************************************************************************************************************
ok: [121.40.237.179] => {
    "msg": "Preparing to start installation, it will take a few minutes."
}

TASK [Install websoft9 when appstore] ***********************************************************************************************************************
changed: [121.40.237.179]

TASK [Replace the key value in system.ini] ******************************************************************************************************************
fatal: [121.40.237.179]: FAILED! => {"changed": false, "msg": "Destination /var/lib/docker/volumes/websoft9_apphub_config/_data/system.ini does not exist !", "rc": 257}

PLAY RECAP **************************************************************************************************************************************************
121.40.237.179             : ok=6    changed=3    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0  
```

**Comment by qiuf12** [comment]
```
TASK [desktop : pause] **************************************************************************************************************************************
ok: [8.136.98.173]

TASK [desktop : set_fact] ***********************************************************************************************************************************
ok: [8.136.98.173]

TASK [desktop : set_fact] ***********************************************************************************************************************************
ok: [8.136.98.173]

TASK [desktop : debug] **************************************************************************************************************************************
ok: [8.136.98.173] => {
    "msg": "Now start to install os_desktop is gnome"
}

TASK [desktop : Ensure the /data/logs directory exists] *****************************************************************************************************
changed: [8.136.98.173]

TASK [desktop : Install Desktop on RedHat] ******************************************************************************************************************

TASK [desktop : Disable firewalld(some cloud has no firewalld)] *********************************************************************************************
changed: [8.136.98.173]

TASK [desktop : Install Desktop and VNC] ********************************************************************************************************************
included: /tmp/20241210085205-appstore/ansible/roles/desktop/tasks/desktop.yml for 8.136.98.173

TASK [desktop : Install GNOME Desktop] **********************************************************************************************************************
changed: [8.136.98.173]

TASK [desktop : Install VNC] ********************************************************************************************************************************
included: /tmp/20241210085205-appstore/ansible/roles/desktop/tasks/vnc.yml for 8.136.98.173

TASK [desktop : Install VNC Server on RedHat] ***************************************************************************************************************
fatal: [8.136.98.173]: FAILED! => {"changed": false, "msg": "The Python 2 yum module is needed for this module. If you require Python 3 support use the `dnf` Ansible module instead."}

PLAY RECAP **************************************************************************************************************************************************
8.136.98.173               : ok=20   changed=9    unreachable=0    failed=1    skipped=34   rescued=0    ignored=0  
```
